awxkit/awxkit/api/pages/base.py

```diff
@@ -177,6 +180,21 @@ def get_oauth2_token(self, username='', password='', client_id=None, description
         else:
             raise exception_from_status_code(resp.status_code)
 
+    def get_oauth2_token(self, username='', password='', client_id=None, description='AWX CLI', client_secret=None, scope='write'):
+        default_cred = config.credentials.default
+        username = username or default_cred.username
+        password = password or default_cred.password
+        # Try gateway first, fallback to controller
+        urls: AuthUrls = {"access_token": "/o/token/", "personal_token": f"{config.gateway_base_path}v1/tokens/"}
+        try:
+            return self._request_token(urls, username, password, client_id, description, client_secret, scope)
+        except exc.NotFound:
+            urls: AuthUrls = {
```

Comment:
Since the type is already declared in the outer scope, it shouldn't be necessary to override it here with a duplicate:
```suggestion
            urls = {
```

Also, could you cover this branch with tests?

Reply:
Both of these branches are covered. That's exactly what the tests I added do.

Reply:
Oh, so coveragepy must be misconfigured somehow: https://app.codecov.io/gh/ansible/awx/pull/15576?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=ansible#049d456459d12f4f8bb1fa1a53f5f733-R192.

I've been meaning to look into it deeper but haven't gotten to that.